docs/handling-eviction-requests.md

```diff
@@ -0,0 +1,79 @@
+| LiveMigrateIfPossible | False             | False                      | True                          | 200 - Eviction granted |
+| External              | N/A               | True                       | True                          | 200 - Eviction granted |
+
+The webhook sets the value of the VMI's `Status.EvacuationNodeName` to the node name its currently running on, so the evacuation controller will know it needs to migrate it to another node.
+
+# Pod Distribution Budget
+In order for the eviction of VMIs to happen in a controlled manner, KubeVirt protects part of the `virt-launcher` pods with a PDB which blocks eviction requests.
+
+`virt-controller` has a `Disruption Budget Controller` which decides whether a `virt-launcher` pod should be protected based on the eviction strategy of its controlling VMI:
+
+| Eviction Strategy     | Is a PDB required             | 
+|-----------------------|-------------------------------|
+| None                  | False                         |
+| LiveMigrate           | True                          |
+| LiveMigrateIfPossible | Only if the VMI is migratable |
+| External              | True                          |
+
+# Eviction Approval Summery
+
+The eviction request's initiator will observe one of the following responses:
+
+| Eviction Strategy     | Is VMI migratable | Is VMI marked for eviction | Does Webhook approve eviction | Does PDB allow eviction | Response                         |
+|-----------------------|-------------------|----------------------------|-------------------------------|-------------------------|----------------------------------|
+| None                  | N/A               | False                      | True                          | True                    | 200 - Eviction granted           |
+| LiveMigrate           | True              | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+| LiveMigrate           | False             | False                      | False                         | False                   | 429 - Eviction denied by webhook |
+| LiveMigrateIfPossible | True              | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+| LiveMigrateIfPossible | False             | False                      | True                          | True                    | 200 - Eviction granted           |
+| External              | N/A               | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+
+To summarize:
+1. The eviction request is granted only if both the webhook and the PDB allow them.
+2. The eviction request's initiator might get a 429 response, but the VMI will be migrated in the background.
+
+# Evacuation Controller
```

Comment:
Why are we using two terms? Is there a difference between Evacuation and Eviction?

Reply:
The name `Evacuation controller` is AFAIU historic.
It is also present in [VMI.status.evacuationNodeName](https://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachineinstancestatus).

In KubeVirt CR [spec.configuration](https://kubevirt.io/api-reference/main/definitions.html#_v1_kubevirtconfiguration), we have `evictionStrategy`.

So it seems that unfortunately there are two names which refer to the same process.
IMO this naming issue deserves a dedicated discussion.
WDYT?

Reply:
I believe that this doc should provide the explanation you just gave.

Reply:
After another reading of the PR description of PR #2091, it seems that both terms are there from day one.